main/boofcv-io/src/main/java/boofcv/io/recognition/RecognitionIO.java

```diff
@@ -845,11 +845,19 @@ public static File downloadModel( List<String> addresses, File destination ) {
 
 	private static void decompressZip( File src, File dst, boolean delete ) {
 		try (java.util.zip.ZipFile zipFile = new ZipFile(src)) {
-			Enumeration<? extends ZipEntry> entries = zipFile.entries();
+			String targetPath = dst.getCanonicalPath();
+      
+      Enumeration<? extends ZipEntry> entries = zipFile.entries();
 			while (entries.hasMoreElements()) {
 				ZipEntry entry = entries.nextElement();
 				File entryDestination = new File(dst, entry.getName());
-				if (entry.isDirectory()) {
+				
+        // Security check to prevent zip slip vulnerability
+        if (!entryDestination.getCanonicalPath().startsWith(targetPath)) {
```

Comment:
I did not know a zip file could reference a parent path? I'll look at your references.